### ISSUE TITLE
Option to set padding

### ISSUE THREAD

**Issue by kepi** [greshake/i3status-rust#377]
Is there any way how to add padding (left, right and between icon and text) to blocks?

![i3status-rust-want-padding](https://user-images.githubusercontent.com/105303/57388715-0fd4bf00-71b9-11e9-8f3d-4364ed9b0f3d.png)

IMHO it is more aesthetic to have padding around icons and between block. My monitor is wide enough and I'm not sure why icons should be so close to text.

**Comment by atheriel** [comment]
It is not currently possible to do this for blocks without a `format` key.

**Comment by ammgws** [comment]
A few blocks have had `format` keys added since this issue was first raised. Also, you should be able to override the icons and add the spaces yourself. Otherwise, not sure what else we can do.

**Comment by kepi** [comment]
> A few blocks have had `format` keys added since this issue was first raised. Also, you should be able to override the icons and add the spaces yourself. Otherwise, not sure what else we can do.

If this was meant like "we don't want to fix it", I'm ok with that. But if this isn't the case and you really are not sure what to do, than I probably explained the issue properly.

We simply need new option on level of  `[theme]` or something similar in "global" scope to change padding of every icon. It would be best to be able to specify left and right padding or margin (not sure about implementation) for all icons at once.

I really can't image why would I override the icons myself. I'm using a computer to do things for me, not other way :) And I'm aware that it would be faster to override every icon in my config (that means find what is original icon, add it to config and add some space...) than to create this feature request. But IMHO there are others who wants this and its not OK to waste so many men-hours on something which can be solved (if it is doable).

I originally talked also about padding between blocks, but it can be fixed with them override in separator, so not as needed as icon padding/margin.

Something like this would be great.

```toml
[theme.override]
icons-padding-right: 5px
icons-padding-left: 10px
```

**Comment by GladOSkar** [comment]
@kepi We generally don't have pixel-perfect control over the bar display since the i3bar protocol is entirely text-based. Adding spaces to icons is the way padding is usually done in the current codebase.

We could _maybe_ (ab)use the `separator_block_width` from the i3bar protocol or some weird pango formatting for this, but those could become messy and might mess with the background colors.

**Comment by ammgws** [comment]
#981 might be useful if "icon_fmt" is introduced

**Comment by kepi** [comment]
@GladOSkar thanks for detailed explanation. I can think about some workarounds if we don't have pixel perfect control like the number would be number of spaces. But I believe that #981 @ammgws mentioned would be easier to implement and probably even simpler to setup. It would also resolve the case with icons inside widget text you mentioned.

Thanks again too both of you for information. I'll subscribe to #981 and hope that `icon_fmt` is introduced some day.

**Comment by ammgws** [comment]
I believe this should is solved since #1095 and #1112 were merged.
